main.go

```diff
@@ -12,7 +14,11 @@ import (
 
 func main() {
 	l := logger.NewLogger()
-	r := repository.NewRedisRepo()
+	r := repository.NewRedisRepo(context.Background(), l, &redis.Options{
+		Addr:     os.Getenv("REDIS_HOST"),
+		Username: os.Getenv("REDIS_USER"),
+		Password: os.Getenv("REDIS_PW"),
```

Comment:
Since [`os.Getenv`](https://pkg.go.dev/os#Getenv) returns an empty string when the value is unset, I think a quick win (and to aid testing locally) is to have a `LookupEnv(key, default) string { ... }` (or equivalent) function which can return the default specified value when the key isn't present.

It's quite a small thing though.